### ISSUE TITLE
Browser Not Supported After Logging In

### ISSUE THREAD

**Issue by Kandelai** [mirkobrombin/Amusiz#8]
It appears they have must of changed some browser configs or requirements. I am getting the following error after entering my login credentials;

Browser Not Supported
We have upgraded and this browser version is no longer supported. Please update your browser.

**Comment by mirkobrombin** [comment]
What distribution and package are you using? 
e.g. Ubuntu with snap?

**Comment by Kandelai** [comment]
I am on EndeavourOS/Arch, using the most current version from AUR, 1.3.4-1. 

My system specs;
Operating System: EndeavourOS
KDE Plasma Version: 5.22.3
KDE Frameworks Version: 5.84.0
Qt Version: 5.15.2
Kernel Version: 5.12.15-zen1-1-zen (64-bit)
Graphics Platform: X11
Processors: 6 × Intel® Core™ i5-9600K CPU @ 3.70GHz
Memory: 15.6 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 1660 SUPER/PCIe/SSE2

**Comment by mirkobrombin** [comment]
It must be a problem with some dependencies, the snap and flatpak are working. I'll set up a vm with arch as soon as possible to test

**Comment by Kandelai** [comment]
Ok, sounds good. thanks.

**Comment by praxishumanus** [comment]
Confirming with the same issue.

Operating System: Manjaro Linux
KDE Plasma Version: 5.21.5
KDE Frameworks Version: 5.82.0
Qt Version: 5.15.2
Kernel Version: 5.10.42-1-MANJARO
OS Type: 64-bit
Graphics Platform: X11
Processors: 4 × Intel® Core™ i5-7500 CPU @ 3.40GHz
Memory: 7.6 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 1650/PCIe/SSE2

**Comment by mirkobrombin** [comment]
Both Arch distributions, I am increasingly convinced it is a dependencies related problem. I am configuring the vm now
Edit: It started appearing on Ubuntu with snap too, I'm afraid it's an Amazon update ..

**Comment by mirkobrombin** [comment]
It started appearing on Ubuntu with snap too, I'm afraid it's an Amazon update ..

**Comment by Florianb63** [comment]
Hello, I have the same problem on my Fedora 34.

**Comment by mirkobrombin** [comment]
I am analyzing the problem and I think it is DRM .. if so, unfortunately there is no solution other than to use chromium embeded or electron (I would avoid).

**Comment by Kandelai** [comment]
This same thing happened with the Nuvola and specifically their add-on; 'nuvola-app-amazon-cloud-player,' causing them to remove Amazon Music as an option. I can't remember what the exact underlining issue was that Amazon changed, but they could not fix it without rewriting their code, something they said is in the works, but won't be out until December.

**Comment by mirkobrombin** [comment]
Nuvola switched to Qt to partially solve the problem if I'm not wrong

**Comment by Kandelai** [comment]
I think they are working on switching to Qt, not sure if they have switched yet, but I just tried Nuvola/AmazonMusic app today and it is still not working, gets the same 'Update Browser' error that this app gets.

**Comment by mirkobrombin** [comment]
Qt might actually be the solution: https://doc.qt.io/archives/qtextended4.4/qtdrmmodule.html
Another solution can be using cefpython: https://github.com/cztomczak/cefpython

**Comment by Kandelai** [comment]
Yes, pretty sure Qt is why Nuvola is switching, but they told me it wouldn't be ready until end of this year.

**Comment by mirkobrombin** [comment]
It's a heavy job for a small application like mine, I can imagine a big one like theirs.

I'm doing some tests with cefpython and if that doesn't work, I'll consider Qt.

**Comment by Kandelai** [comment]
Sounds good. I will monitor your progress and check back. Thanks again, damn DRM.

**Comment by mirkobrombin** [comment]
![image](https://user-images.githubusercontent.com/9479903/125519984-b53b6c98-f11b-4146-8121-541c60dfb8eb.png)
Unfortunately initial tests with QWebEngineView failed.

**Comment by mirkobrombin** [comment]
![image](https://user-images.githubusercontent.com/9479903/125522188-9a3d270c-d834-4830-899e-f9f241cdeac1.png)
Fun fact, Amazon Music for Windows start working with the latest build of another my application :D

**Comment by Suzie97** [comment]
If it a dependency issue, flatpak packaging should solve it. So is providing other native packaging options a necessity?

**Comment by mirkobrombin** [comment]
> If it a dependency issue, flatpak packaging should solve it. So is providing other native packaging options a necessity?

It is not a dependencies problem. Amazon has made some changes to the platform.
The cause should be DRM incompatibility with webkit2gtk. I am evaluating other solutions but at the moment I don't know what to do.

I am considering creating a snap/flatpak/appimage of the Windows version

**Comment by j0chn1** [comment]
Hey there,
I freshly installed Linux and Amusiz and got the same message / error.
![image](https://user-images.githubusercontent.com/87828588/126690363-d844da93-8185-49b5-846e-eb797f57166a.png)

**Comment by mirkobrombin** [comment]
> Hey there,
> I freshly installed Linux and Amusiz and got the same message / error.
> ![image](https://user-images.githubusercontent.com/87828588/126690363-d844da93-8185-49b5-846e-eb797f57166a.png)

https://github.com/mirkobrombin/Amusiz/issues/8#issuecomment-879745120

**Comment by j0chn1** [comment]
> > Hey there,
> > I freshly installed Linux and Amusiz and got the same message / error.
> [#8 (comment)](https://github.com/mirkobrombin/Amusiz/issues/8#issuecomment-879745120)

![image](https://user-images.githubusercontent.com/87828588/126700241-1c0b9e23-162b-442b-b0d8-8f293af40d6f.png)

[🔴] × flatpak-builder --repo=amusiz --force-clean --user build-dir pm.mirko.Amusiz.yml
Can't parse 'pm.mirko.Amusiz.yml': 114:65: mapping values are not allowed in this context

**Comment by mirkobrombin** [comment]
> > > Hey there,
> > > I freshly installed Linux and Amusiz and got the same message / error.
> > > [#8 (comment)](https://github.com/mirkobrombin/Amusiz/issues/8#issuecomment-879745120)
> 
> ![image](https://user-images.githubusercontent.com/87828588/126700241-1c0b9e23-162b-442b-b0d8-8f293af40d6f.png)
> 
> [] × flatpak-builder --repo=amusiz --force-clean --user build-dir pm.mirko.Amusiz.yml
> Can't parse 'pm.mirko.Amusiz.yml': 114:65: mapping values are not allowed in this context

Yeah ok but Amusiz is currently killed by Amazon and DRM .-. I need to take my time to re-project and rebuild the project using different technologies, at least a toolkit that provide a webview DRM capable.

**Comment by j0chn1** [comment]
> Yeah ok but Amusiz is currently killed by Amazon and DRM .-. I need to take my time to re-project and rebuild the project using different technologies, at least a toolkit that provide a webview DRM capable.

Sad to hear so. But I am looking forward for a new version. 
I appreciate your work. Thank you!

**Comment by ZeroDot1** [comment]
Maybe the problem with the incompatible browser could be solved by simple user agent spoofing.

**Comment by mirkobrombin** [comment]
No, already tried
